src/newt/db/search.py

```diff
@@ -173,6 +173,10 @@ def create_text_index(conn, fname, D, C=None, B=None, A=None):
         conn.commit()
     finally:
         try:
+            cursor.close()
```

Comment:
In most DBAPI implementations I've looked at (far too many 😄 ), the cursor has a reference to the connection, but the connection does not have references to the cursors that are opened.

Now, closing the connection will close the socket, it's true. And that would probably be enough to solve this particular case. In general, however, it's best to be sure to close the cursors, I've found. A buffered cursor (most of them are by default) will have the last row set in memory and it won't get rid of it until its closed (or GCd), so closing it allows resources to be cleaned up sooner and deterministically. With unbuffered cursors, if you don't close the cursor then when you close the connection it will detect that it was in the middle of a result set and complain either by logging  a warning (pymysql) or raising an error (Oracle's official driver). I haven't tried unbuffered cursors on postgres so I don't know what they do.